src/ucs/sys/string.c

```diff
@@ -267,3 +268,26 @@ const char * ucs_str_dump_hex(const void* data, size_t length, char *buf,
     *p = 0;
     return buf;
 }
+
+const char* ucs_flags_str(char *buf, size_t max,
+                          const void *src, const void *arg)
+{
+    char * const *table;
+    int i, len;
+
+    len = 0;
+    for (table = arg, i = 0; *table; ++table, ++i) {
```

Comment:
pls use ucs_for_each_bit
if we iterate on bits, not on table entries, would not need to set NULL as the last table entry..

Reply:
I had ucs_for_each_bit in the previous function but i thought you meant to keep the code as it was in parser.c (without ucs_for_each_bit).
I guess i misunderstood, I'll restore.

Reply:
@yosefe the way it is now, with ucs_for_each_bit: 
```
ucs_for_each_bit(i, flags) {
        snprintf(buf + len, max - len, "%s,", str_table[i]);
        len = strlen(buf);
}
```
makes coverity complain about:
```
Event overrun-buffer-val: 	Overrunning array "ucp_feature_str" of 56 bytes by passing it to a function which accesses it at byte offset 504.
```
because inside ucs_flags_str,:
```
Event index_bounded_by_const: 	Pointer "str_table" is indexed by "i", whose upper bound is constant "64" in "i < 64".
```

To fix this issue, I can change it back to:
```
    for (i = 0; *str_table; ++str_table, ++i) {
        if (flags & UCS_BIT(i)) {
            snprintf(buf + len, max - len, "%s,", *str_table);
            len = strlen(buf);
        }
    }
```
and add NULL at the tables' end.
is that ok?

Reply:
ok , add comments it's because of coverity